release-20.2: delegate: fix zone configs for SHOW CREATE TABLE with no partitions

otan opened the pull request:
Backport commits from #65167 and #61235.

/cc @cockroachdb/release

---

See individual PRs for details.

otan committed:
sql: fix bug where zone config gets created on ALTER PRIMARY KEY

Release justification: fix high-severity bug in existing functionality

Release note (bug fix): Fix a case where empty zone configurations get
created for certain indexes during ALTER PRIMARY KEY.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/65175)
<!-- Reviewable:end -->

otan committed:
delegate: fix zone configs for SHOW CREATE TABLE with no partitions

Fixed the regression in c44619c38ba6f22d8d6590a34600ba0c2fd7fdd9 which
had no tests to cover this case.

Release note (bug fix): Fixed a bug where if the zone configurations
would not display on a SHOW CREATE TABLE if there are no partitions
even though there are zone configurations on the index or tables
themselves.
